tests/system/Cache/Handlers/RedisHandlerTest.php

```diff
@@ -0,0 +1,117 @@
+<?php namespace CodeIgniter\Cache\Handlers;
+
+class RedisHandlerTest extends \CIUnitTestCase
+{
+	private $redisHandler;
+	private static $key1 = 'key1';
+	private static $key2 = 'key2';
+	private static $key3 = 'key3';
+	private static function getKeyArray()
+	{
+		return [
+			self::$key1, self::$key2, self::$key3
+		];
+	}
+
+	private static $dummy = 'dymmy';
+	private $config;
+
+	public function setUp()
+	{
+		$this->config = new \Config\Cache();
+
+		$this->redisHandler = new RedisHandler($this->config->redis);
+		if (!$this->redisHandler->isSupported()) {
+			$this->markTestSkipped('Not support redis');
+		}
+
+		$this->redisHandler->initialize();
+	}
+
+	public function tearDown()
+	{
+		foreach (self::getKeyArray() as $key) {
+			$this->redisHandler->delete($key);
+		}
+	}
+
+	public function testNew()
+	{
+		$this->assertInstanceOf(RedisHandler::class, $this->redisHandler);
+	}
+
+	public function testDestruct()
+	{
+		$this->redisHandler = new RedisHandler($this->config->redis);
+		$this->redisHandler->initialize();
+
+		$this->assertInstanceOf(RedisHandler::class, $this->redisHandler);
+	}
+
+
+	public function testGet()
+	{
+		$this->redisHandler->save(self::$key1, 'value', 1);
+
+		$this->assertSame('value', $this->redisHandler->get(self::$key1));
+		$this->assertFalse($this->redisHandler->get(self::$dummy));
+
+		\CodeIgniter\CLI\CLI::wait(2);
+		$this->assertFalse($this->redisHandler->get(self::$key1));
+	}
+
+	public function testSave()
+	{
+		$this->assertTrue($this->redisHandler->save(self::$key1, 'value'));
+	}
+
+	public function testDelete()
+	{
+		$this->redisHandler->save(self::$key1, 'value');
+
+		$this->assertTrue($this->redisHandler->delete(self::$key1));
+		$this->assertFalse($this->redisHandler->delete(self::$dummy));
+	}
+
+	//FIXME: I don't like all Hash logic very much. It's wasting memory.
```

Comment:
Redis strcture is following now.  So 'data' field is not exists.
```
'key' : [
    '__ci_type' : xxxxx,
    '__ci_value': yyyyy
]
```
And I think Redis have many functions. I want to use ```set```, ```get``` and etc too.
```set``` is very simple and a small amount of memory.